static/app/components/events/eventViewHierarchy.tsx

```diff
@@ -0,0 +1,95 @@
+import {useState} from 'react';
+import isEqual from 'lodash/isEqual';
+
+import {getAttachmentUrl} from 'sentry/components/events/attachmentViewers/utils';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {tn} from 'sentry/locale';
+import {EventAttachment} from 'sentry/types';
+import {uniqueId} from 'sentry/utils/guid';
+import {useQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+import EventDataSection from './eventDataSection';
+import {ViewHierarchy, ViewHierarchyData} from './viewHierarchy';
+
+const DEFAULT_RESPONSE: ViewHierarchyData = {rendering_system: '', windows: []};
+const FIVE_SECONDS_IN_MS = 5 * 1000;
+
+function fillWithUniqueIds(hierarchy) {
+  return {
+    ...hierarchy,
+    id: uniqueId(),
+    children: hierarchy.children?.map(fillWithUniqueIds) ?? [],
+  };
+}
+
+type Props = {
+  projectSlug: string;
+  viewHierarchies: EventAttachment[];
+};
+
+function EventViewHierarchy({projectSlug, viewHierarchies}: Props) {
+  const [selectedViewHierarchy] = useState(0);
+  const api = useApi();
+  const organization = useOrganization();
+
+  const hierarchyMeta = viewHierarchies[selectedViewHierarchy];
+  const {isLoading, data} = useQuery<ViewHierarchyData>(
+    [
+      getAttachmentUrl({
+        attachment: hierarchyMeta,
+        eventId: hierarchyMeta.event_id,
+        orgId: organization.slug,
+        projectId: projectSlug,
+      }),
```

Comment:
do we need to get this whole attachment url here?

Reply:
Replaced with just `viewHierarchies.${hierarchyMeta.id}`! This should be simple enough to act as a cache key